src/node/internal/crypto.d.ts

```diff
@@ -2,4 +2,100 @@
 // Licensed under the Apache 2.0 license found in the LICENSE file or at:
 //     https://opensource.org/licenses/Apache-2.0
 
-export function getRandomInt(min: number, max: number): number;
+export function exportKey(key: CryptoKey, options?: ExportOptions): string|Uint8Array|JsonWebKey;
+export function equals(key: CryptoKey, otherKey: CryptoKey): boolean;
+export function getAsymmetricKeyDetail(key: CryptoKey): AsymmetricKeyDetails;
+export function getAsymmetricKeyType(key: CryptoKey): AsymmetricKeyType;
+export function generateKey(type: 'hmac' | 'aes', options: GenerateKeyOptions): CryptoKey;
+export function generateKeyPair(type: AsymmetricKeyType, options: GenerateKeyPairOptions): CryptoKeyPair;
+export function createSecretKey(key: string | ArrayBuffer | Buffer | ArrayBufferView, encoding?: string): CryptoKey;
+export function createPrivateKey(key: CreatePrivateKeyOptions |
+                                      string |
+                                      ArrayBuffer |
+                                      ArrayBufferView) : CryptoKey;
+export function createPublicKey(key: CreatePublicKeyOptions |
+                                     string |
+                                     ArrayBuffer |
+                                     ArrayBufferView) : CryptoKey;
+
+import { Buffer } from 'node-internal:internal_buffer';
+
+export interface SecretKeyExportOptions {
+  format?: 'buffer' | 'jwk';
+}
+
+export interface PublicKeyExportOptions {
+  type?: 'pkcs1' | 'spki';
+  format?: 'pem' | 'der' | 'jwk';
+}
+
+export interface PrivateKeyExportOptions {
+  type?: 'pkcs1' | 'pkcs8' | 'sec1';
+  format?: 'pem' | 'der' | 'jwk';
+  cipher?: string;
+  passphrase?: string | Uint8Array;
+}
+
+export type ExportOptions = SecretKeyExportOptions |
+                            PublicKeyExportOptions |
+                            PrivateKeyExportOptions;
+
+export interface AsymmetricKeyDetails {
+  modulusLength?: number;
+  publicExponent?: bigint;
+  hashAlgorithm?: string;
+  mgf1HashAlgorithm?: string;
+  saltLength?: number;
+  divisorLength?: number;
+  namedCurve?: string;
+}
+
+export enum AsymmetricKeyType {
+  RSA = 'rsa',
+  RSA_PSS = 'rsa-pss',
+  DSA = 'dsa',
+  EC = 'ec',
+  X25519 = 'x25519',
+  X448 = 'x448',
```

Comment:
X448 and ED448 should not be needed here

Reply:
I wouldn't spend time reviewing this just yet. This is still very much an active WIP.